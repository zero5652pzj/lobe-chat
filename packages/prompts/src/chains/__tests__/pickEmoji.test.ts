import { describe, expect, it } from 'vitest';

import { chainPickEmoji } from '../pickEmoji';

describe('chainPickEmoji', () => {
  it('should return a payload with the correct structure and embedded user content', () => {
    const result = chainPickEmoji('你是一名星际探索者');

    expect(result).toMatchSnapshot();
  });
});
