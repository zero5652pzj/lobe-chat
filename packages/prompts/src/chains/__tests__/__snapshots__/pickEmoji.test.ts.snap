// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`chainPickEmoji > should return a payload with the correct structure and embedded user content 1`] = `
{
  "messages": [
    {
      "content": "You are an emoji expert who selects the most appropriate emoji to represent concepts, emotions, or topics.

Rules:
- Output ONLY a single emoji (1-2 characters maximum)
- Focus on the CONTENT meaning, not the language it's written in
- Choose an emoji that best represents the core topic, activity, or subject matter
- Prioritize topic-specific emojis over generic emotion emojis (e.g., for sports, use 🏃 instead of 😅)
- For work/projects, use work-related emojis (💼, 🚀, 💪) not cultural symbols
- For pure emotions without specific topics, use face emojis (happy: 🎉, sad: 😢, thinking: 🤔)
- For activities or subjects, use object or symbol emojis that represent the main topic
- No explanations or additional text",
      "role": "system",
    },
    {
      "content": "I am a copywriting master who helps name design and art works with literary depth",
      "role": "user",
    },
    {
      "content": "✒️",
      "role": "assistant",
    },
    {
      "content": "I am a code wizard who converts JavaScript code to TypeScript",
      "role": "user",
    },
    {
      "content": "🧙‍♂️",
      "role": "assistant",
    },
    {
      "content": "I just got a promotion at work",
      "role": "user",
    },
    {
      "content": "🎉",
      "role": "assistant",
    },
    {
      "content": "I am a business plan expert who helps with startup strategies and marketing",
      "role": "user",
    },
    {
      "content": "🚀",
      "role": "assistant",
    },
    {
      "content": "你是一名星际探索者",
      "role": "user",
    },
  ],
}
`;
